### PR TITLE
Update http example with json protocol

### PR DESCRIPTION
**Pull request by ramgdev**
Fixes #
Design discussion issue (if applicable) #

## Changes

- Update http example to enable `http-json` feature 
- Update code to specify `protocol` when building http exporter

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Review by lalitb — APPROVED** [review]
Thanks, LGTM.

File: opentelemetry-otlp/CHANGELOG.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ now use `.with_resource(RESOURCE::default())` to configure Resource when using
   previous release.
 - **Breaking** [1869](https://github.com/open-telemetry/opentelemetry-rust/pull/1869) The OTLP logs exporter now overrides the [InstrumentationScope::name](https://github.com/open-telemetry/opentelemetry-proto/blob/b3060d2104df364136d75a35779e6bd48bac449a/opentelemetry/proto/common/v1/common.proto#L73) field with the `target` from `LogRecord`, if target is populated.
 - Groups batch of `LogRecord` and `Span` by their resource and instrumentation scope before exporting, for better efficiency [#1873](https://github.com/open-telemetry/opentelemetry-rust/pull/1873).
-
+- Updated basic-otlp-http example to make use of `protocol` setting when building the exporter
```
Comment (cijothomas) [inline opentelemetry-otlp/CHANGELOG.md:23]: not a product change, so no need of changelog.

File: opentelemetry-otlp/examples/basic-otlp-http/src/main.rs [hunk 2]
```diff
@@ -39,22 +39,34 @@ fn init_logs() -> Result<sdklogs::LoggerProvider, opentelemetry::logs::LogError>
...
         .install_batch(opentelemetry_sdk::runtime::Tokio)
 }
 
 fn init_tracer_provider() -> Result<sdktrace::TracerProvider, TraceError> {
     opentelemetry_otlp::new_pipeline()
         .tracing()
-        .with_exporter(http_exporter().with_endpoint("http://localhost:4318/v1/traces"))
+        .with_exporter(
+            http_exporter()
+                .with_protocol(Protocol::HttpBinary)
```
Comment (cijothomas) [inline opentelemetry-otlp/examples/basic-otlp-http/src/main.rs:55]: I am not sure if I follow the PR's intent.. The title says it is updating example to show json payload, but actual changes does not indicate so.. Could you please clarify the intent?

Comment (lalitb) [reply opentelemetry-otlp/examples/basic-otlp-http/src/main.rs:55]: I think we can just add a comment here (and Readme.md) on how to switch protocol from `HttpBinary` to `HttpJson`. And default can remain binary.

**Pull request merged**